libwire/inc/IpHeader.h

```diff
@@ -1,55 +1,62 @@
 #ifndef IPHEADER_H
 #define IPHEADER_H
 
-#include <iostream>
+#include <string>
+
 #include <stdint.h>
 #include <asm/types.h>
 #include <linux/posix_types.h>
 #include <asm/byteorder.h>
 
+#include "NetUtils.h"
+#include "Utils.h"
+
 namespace Net {
 
-	struct iphdr {
-	#if defined(__LITTLE_ENDIAN_BITFIELD)
-		__u8	ihl:4,
-		version:4;
-	#	elif defined (__BIG_ENDIAN_BITFIELD)
-		__u8	version:4,
-		ihl:4;
-	#else
-	#error	"Please fix <asm/byteorder.h>"
-	#endif
-	__u8	tos;
-	__be16	tot_len;
-	__be16	id;
-	__be16	frag_off;
-	__u8	ttl;
-	__u8	protocol;
-	__sum16	check;
-	__be32	saddr;
-	__be32	daddr;
-	/*The options start here. */
-	};
+    typedef struct iphdr {
+        #if defined(__LITTLE_ENDIAN_BITFIELD)
+            __u8	ihl:4,
+            version:4;
+        #	elif defined (__BIG_ENDIAN_BITFIELD)
+            __u8	version:4,
+            ihl:4;
+        #else
+        #error	"Please fix <asm/byteorder.h>"
+        #endif
+        __u8	tos;
+        __be16	tot_len;
+        __be16	id;
+        __be16	frag_off;
+        __u8	ttl;
+        __u8	protocol;
+        __sum16	check;
+        __be32	saddr; // Does is work for ipv6 addresses?
+        __be32	daddr;
+    } iphdr_t;
 	
 	class IpHeader
 	{
-	private:
-		struct iphdr *ip_header;
 	public:
-		explicit IpHeader(unsigned char *buffer);
+		explicit IpHeader(data_t const& buffer);
```

Comment:
Use data_t type (which is a typedef on std::string),
so we can use size() and other method on it